src/vs/editor/contrib/inlineCompletions/browser/inlineCompletionsModel.ts

```diff
@@ -486,6 +469,34 @@ export class InlineCompletionsModel extends Disposable {
 	}
 }
 
+export function getSecondaryEdits(textModel: ITextModel, positions: readonly Position[], primaryEdit: SingleTextEdit): SingleTextEdit[] {
+	const primaryPosition = positions[0];
+	const secondaryPositions = positions.slice(1);
+	const primaryEditEndPosition = primaryEdit.range.getEndPosition();
+	const replacedTextAfterPrimaryCursor = textModel.getValueInRange(
+		Range.fromPositions(primaryPosition, primaryEditEndPosition)
+	);
+	const positionWithinTextEdit = subtractPositions(primaryPosition, primaryEdit.range.getStartPosition());
+	const secondaryEditText = getTextFromPosition(primaryEdit.text, positionWithinTextEdit);
+	return secondaryPositions.map(pos => {
+		const textAfterSecondaryCursor = textModel.getValueInRange(
+			Range.fromPositions(pos, primaryEditEndPosition)
+		);
+		const l = commonPrefixLength(replacedTextAfterPrimaryCursor, textAfterSecondaryCursor);
+		const range = Range.fromPositions(pos, pos.delta(0, l));
+		return new SingleTextEdit(range, secondaryEditText);
+	});
+}
+
+function getTextFromPosition(text: string, pos: Position): string {
+	let subtext = '';
+	const { lines, separators } = splitLinesIncludeSeparators(text);
+	for (let i = pos.lineNumber - 1; i < lines.length; i++) {
+		subtext += lines[i].substring(i === pos.lineNumber - 1 ? pos.column - 1 : 0) + (separators[i] ?? '');
```

Comment:
Do you really always want to add the separator?
I think it might be easier if `splitLinesIncludingSeparators` doesn't have the separators separately, but leaves them in the lines, e.g.:
`"line1\nline2\r\nlast" -> ["line1\n", "line2\r\n", "last"]`

Reply:
Yes sounds good, I combined them